packages/aws-cdk/lib/toolkit/cli-io-host.ts

```diff
@@ -21,7 +21,21 @@ interface IoMessage {
   readonly action: IoAction;
 
   /**
-   * A short code uniquely identifying message type.
+   * A short message code uniquely identifying a message type of the format [CATEGORY]_[NUMBER_CODE].
+   * All 'error' level message codes begin with 0.
+   * All warning level message codes begin with 1.
+   * All info level message codes begin with 2.
+   *
+   * Codes ending in 000 are generic messages, while codes ending in 001-999 are specific to a particular message.
```

Comment:
I don't fully understand the need for this. Isn't every code attached to a particular message? 

Can you show an example of both authoring and consuming different codes here?

I feel like we can just say that we have 3 digits to denote the message and thats it, without the special handling of `000`. Also having a generic code might eventually result in a graveyard (though not having it will bloat up the numbers). 

We can also start without `000`. That is, start with `001...999` and add `000` as a "generic" thing if we need it.

Reply:
Every message will have a code but they do not need to be a unique code. I fully suspect that especially for `info` level codes that people really won't care to differentiate as much as they would with warnings or errors. I suspect any arbitrary log message today that is like "doing <thing>, please wait" probably doesn't warrant a code so it'll be a generic info code. The alternative is that code becomes an optional value (which is probably fine), and is really only relevant for errors and warnings